app/src/main/java/ca/rmen/android/poetassistant/main/MainActivity.java

```diff
@@ -149,8 +149,14 @@ protected void onNewIntent(Intent intent) {
         setIntent(intent);
         // The user entered a search term either by typing or by voice
         if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
-            String query = intent.getStringExtra(SearchManager.QUERY);
-            if (TextUtils.isEmpty(query)) query = intent.getStringExtra(SearchManager.USER_QUERY);
+            String query = intent.getDataString();
+            if (TextUtils.isEmpty(query)) {
+                query = intent.getStringExtra(SearchManager.QUERY);
+            }
+            if (TextUtils.isEmpty(query)) {
+                CharSequence userQuery = intent.getCharSequenceExtra(SearchManager.USER_QUERY);
```

Comment:
Con: from testing, we see that `USER_QUERY` is now a `SpannableStringBuilder` instead of a`String`